src/semanticlayertools/clustering/leiden.py

```diff
@@ -0,0 +1,174 @@
+import os
+import time
+import re
+from tqdm import tqdm
+
+import igraph as ig
+import leidenalg as la
+
+
+class TimeCluster():
+    """Cluster time-sliced data with the Leiden algorithm.
+
+    Calculates temporal clusters of e.g. time-sliced cocitation or citation
+    data, using the Leiden algorithm . Two nodes are assumed to be identical in
+    different year slices, if the node name is the same.
+    This could be e.g. the bibcode or DOI.
+
+    Input files are assumed to include the year in the filename, have an ending
+    `_GC.net` to denote their giant component character and should be in Pajek
+    format.
+
+    The resolution parameter can be seen as a limiting density, above
+    which neighbouring nodes are considered a cluster. The interslice coupling
+    describes the influcence of yearly order on the clustering process. See doc
+    for the Leiden algorithm for more detailed info.
+
+    :param inpath: Path for input network data
+    :type inpath: str
+    :param outpath: Path for writing output data
+    :type outpath: str
+    :param resolution: Main parameter for the clustering quality function (Constant Pots Model)
+    :type resolution: float
+    :param intersliceCoupling: Coupling parameter between two year slices, also influences cluster detection
+    :type intersliceCoupling: float
+    :param timerange: The time range for considering input data (default=1945,2005))
+    :type timerange: tuple
+    :raises OSError: If the output file already exists at class instantiation
+
+    .. seealso::
+       Traag, V.A., Waltman. L., Van Eck, N.-J. (2018).
+       From Louvain to Leiden: guaranteeing well-connected communities.
+       Scientific reports, 9(1), 5233. 10.1038/s41598-019-41695-z
+    """
+
+    def __init__(
+        self, inpath: str, outpath: str,
+        resolution: float = 0.003,
+        intersliceCoupling: float = 0.4,
+        timerange: tuple = (1945, 2005),
+        useGC: bool = True,
```

Comment:
This parameter is not documented, I'm uncertain what it does.

Reply:
Reading further down, it probably means "use Giant Component". This makes sense, but before reading done, I suspect `useGC` had something to do with garbage collection.

Reply:
this is already fixed upstream